File: frontend/src/services/api/progression.ts
// Feature: F003

import apiClient from './client';

export interface BadgeResponse {
  id: number;
  name: string;
  description: string | null;
  icon_url: string | null;
}

export interface ProgressionResponse {
  player_id: number;
  points: number;
  level: number;
  badges: BadgeResponse[];
  tournaments_played: number;
  matches_won: number;
  points_to_next_level: number;
}

export interface PlayerProfileResponse {
  id: number;
  username: string;
  email: string;
  rank: string;
  region: string;
  account_status: string;
  created_at: string;
  points: number;
  level: number;
  badges: BadgeResponse[];
  tournaments_played: number;
  matches_won: number;
  points_to_next_level: number;
}

export const getPlayerProfile = async (playerId: number): Promise<PlayerProfileResponse> => {
  const response = await apiClient.get<PlayerProfileResponse>(`/players/${playerId}/profile`);
  return response.data;
};

export const getAllBadges = async (): Promise<BadgeResponse[]> => {
  const response = await apiClient.get<BadgeResponse[]>('/progression/badges/');
  return response.data;
};

File: backend/app/schemas/progression.py
# Feature: F003
# Scenario: SC005

from pydantic import BaseModel
from typing import List, Optional


class BadgeResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    icon_url: Optional[str] = None

    class Config:
        from_attributes = True


class ProgressionResponse(BaseModel):
    player_id: int
    points: int
    level: int
    badges: List[BadgeResponse]
    tournaments_played: int
    matches_won: int
    points_to_next_level: int

    class Config:
        from_attributes = True
